### PR TITLE
Move JHttpFactory::getHttp() into try catch and add error handling

### PR DESCRIPTION
**Pull request by zero-24**
### Summary of Changes

Move any usage of  `JHttpFactory::getHttp()` into try catch and add error handling

This is a redo from #15388 cc @mbabker 

### Testing Instructions

Review.

Confirm that you can still uses a feed.
Confirm that recaptcha still works.

### Expected result

everything still works

### Actual result

everything works but without error handling

### Documentation Changes Required

None.

### CODE REVIEW

**Review by mbabker — APPROVED** [review]

File: plugins/twofactorauth/yubikey/yubikey.php [hunk 1]
```diff
@@ -368,4 +368,4 @@ public function validateYubikeyOtp($otp)
 
 		return true;
 	}
-}
+}
```
Comment (jeckodevelopment) [inline plugins/twofactorauth/yubikey/yubikey.php]: @zero-24 drone says:
image

**Comment by SamuelSchepp** [comment]
Please provide instructions for testing a failed $reader->open, but a successful HttpFactory::getHttp.
@icampus<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17533">issues.joomla.org/tracker/joomla-cms/17533</a>.</sub>

**Comment by zero-24** [comment]
> Please provide instructions for testing a failed $reader->open, but a successful HttpFactory::getHttp.
@icampus

Sorry i don't understand the request. Just test that pre and after patch the fedd and catcha still works ;) Noting special to test 😄

**Comment by SamuelSchepp** [comment]
I have tested this item :white_check_mark: successfully on 1905a8a419654f7239435a646a50a9409c419d8d<br /><br />Tested as described in "Testing Instructions".
@icampus
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17533">issues.joomla.org/tracker/joomla-cms/17533</a>.</sub>

**Comment by Worti2** [comment]
I have tested this item :white_check_mark: successfully on 5bcbe78a598beb302490a1a7daa3155ff9f28489<br /><br />I tested this patch successfully.
Regards
Markus<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17533">issues.joomla.org/tracker/joomla-cms/17533</a>.</sub>

**Comment by zero-24** [comment]
Thanks @Worti2 👍 

@mbabker can we finally merge this one?

**Comment by zero-24** [comment]
Thanks 👍

**Pull request merged**